pkg/runtime/serializer/versioning/versioning.go

```diff
@@ -102,11 +117,30 @@ func (c *codec) Decode(data []byte, defaultGVK *unversioned.GroupVersionKind, in
 			}
 			return into, gvk, nil
 		}
+
+		// perform defaulting if requested
+		if c.defaulter != nil {
+			// create a copy to ensure defaulting is not applied to the original versioned objects
+			if isVersioned {
+				copied, err := c.copier.Copy(obj)
+				if err != nil {
+					copied = obj
```

Comment:
Now I see it's pretty much copy from the below code, so maybe Errorf in both places...